pkg/controller/certificates/trigger/policies/gatherer.go

```diff
@@ -70,40 +70,60 @@ func (g *Gatherer) DataForCertificate(ctx context.Context, crt *cmapi.Certificat
-	// that the certificate's revision field stays nil until the first
-	// certificate request (revision "1") has become ready.
-	if crt.Status.Revision == nil {
-		return Input{
-			Certificate: crt,
-			Secret:      secret,
-		}, nil
+	// Attempt to fetch the CertificateRequest for the current status.revision.
+	//
+	// We can skip looking for the current CR when the status.revision is nil
+	// since there cannot be any available "current" certificate request if the
+	// certificate's revision is empty. That is due to the fact that the
+	// certificate's revision field stays nil until the first certificate
+	// request (revision "1") has become ready.
+	var curCR *cmapi.CertificateRequest
+	if crt.Status.Revision != nil {
+		reqs, err := certificates.ListCertificateRequestsMatchingPredicates(g.CertificateRequestLister.CertificateRequests(crt.Namespace),
+			labels.Everything(),
+			predicate.ResourceOwnedBy(crt),
+			predicate.CertificateRequestRevision(*crt.Status.Revision),
+		)
+		if err != nil {
+			return Input{}, err
+		}
+		switch {
+		case len(reqs) > 1:
+			return Input{}, fmt.Errorf("multiple CertificateRequest resources exist for the current revision %v, not triggering new issuance until requests have been cleaned up", *crt.Status.Revision)
+		case len(reqs) == 1:
+			curCR = reqs[0]
+		case len(reqs) == 0:
+			log.V(logf.DebugLevel).Info("Found no CertificateRequest resources owned by this Certificate for the current revision", "revision", *crt.Status.Revision)
+		}
 	}
 
-	// Attempt to fetch the CertificateRequest resource for the current
-	// 'status.revision'.
-	var req *cmapi.CertificateRequest
+	// Attempt fetching the CertificateRequest for the next status.revision.
+	var nextCR *cmapi.CertificateRequest
+	nextCRRevision := 1
+	if crt.Status.Revision != nil {
+		nextCRRevision = *crt.Status.Revision + 1
+	}
 	reqs, err := certificates.ListCertificateRequestsMatchingPredicates(g.CertificateRequestLister.CertificateRequests(crt.Namespace),
 		labels.Everything(),
 		predicate.ResourceOwnedBy(crt),
-		predicate.CertificateRequestRevision(*crt.Status.Revision),
+		predicate.CertificateRequestRevision(nextCRRevision),
 	)
 	if err != nil {
 		return Input{}, err
 	}
 	switch {
 	case len(reqs) > 1:
-		return Input{}, fmt.Errorf("multiple CertificateRequest resources exist for the current revision, not triggering new issuance until requests have been cleaned up")
+		return Input{}, fmt.Errorf("multiple CertificateRequest resources exist for the next revision %d, not triggering new issuance until requests have been cleaned up", nextCRRevision)
 	case len(reqs) == 1:
-		req = reqs[0]
+		nextCR = reqs[0]
 	case len(reqs) == 0:
-		log.V(logf.DebugLevel).Info("Found no CertificateRequest resources owned by this Certificate for the current revision", "revision", *crt.Status.Revision)
+		log.V(logf.DebugLevel).Info("Found no CertificateRequest resources owned by this Certificate for the next revision", "revision", nextCRRevision)
 	}
```

Comment:
Is that you had in mind @JoshVanL?

Context: we wanted to be able to fetch the "next" certificate request so that we could compare the certificate's spec with the next certificate request and re-issue immediately if there was a mismatch. I remember we discussed this with @munnerz and you on one of our standups; I remember we talked about the idea of adding a NextRevisionRequest field to DataForCertificate.

Reply:
^ PR is now ready to review 😃 

@JoshVanL I remember you had some thoughts on how to fetch the "next revision CR" (as opposed to the current revision CR). Thoughts?

Reply:
Resolving now, we will be merging soon